tiiuae/ghaf-givc
# Add wifi functionality to systemd server

## Description of changes
wifi server is added to systemd server on net-vm
wifi AP list and connection nmcli queries are added

## Testing

AP List:
- Connect to gui-vm via SSH
- Use grpcurl to query of wifi AP list: `grpcurl -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.ListNetwork`
- Observe the network list with in-use, SSID, signal strength and security lists. 

Connect:
- Connect to gui-vm via SSH
- Replace ssid and password in the command and run grpcurl to connect to AP: `grpcurl '{"SSID":"ssid","Password":"password"}' -cacert /run/givc/ca-cert.pem -cert /etc/givc/gui-vm.ghaf-cert.pem -key /etc/givc/gui-vm.ghaf-key.pem 192.168.101.1:9000 systemd.WifiService.ConnectNetwork`
- Observe the network connection on the Waybar.
